foundation-ui/mount/MountService.js

```diff
@@ -0,0 +1,94 @@
+import {EventEmitter} from 'events';
+import {CHANGE} from './MountEvent';
+
+/**
+ * List of component descriptors
+ *
+ * @type {Array.<{component:function, priority:Number, type:String}>}
+ */
+const components = [];
+
+/**
+ * Consecutive number used to rank components based on order of registration
+ *
+ * @type {number}
+ */
+let rank = 0;
+
+/**
+ * MountService
+ */
+class MountService extends EventEmitter {
+
+  /**
+   * Register component with type and optional priority
+   *
+   * respective mounts
+   * @param {function} component - React component which will be
+   * mounted based on the provided type
+   * @param {String} type - String which is used to match components with the
+   * @param {Number} [priority] - Number which is used to sort the components
+   */
+  registerComponent(component, type, priority = 0) {
+    if (typeof component !== 'function') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided component must be a ' +
+            'React.Component constructor or a stateless functional component.');
+      }
+
+      return;
+    }
+
+    if (typeof type !== 'string' || type === '') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided type must be a none empty string');
+      }
+
+      return;
+    }
+
+    // Add component descriptor and sort components list by priority and rank
+    components.push({component, priority, type, rank: rank++});
+    components.sort((a, b) => {
+      if (a.priority !== b.priority) {
+        return b.priority - a.priority;
+      }
+
+      return a.rank - b.rank;
+    });
```

Comment:
This makes sense. I was getting thrown off with what the most important priority was, as I was thinking the lowest would be most important, but that's because I'm comparing it to the priorities in our Hooks.
I'm cool with leaving it as it is, but it's worth considering that it's different between the two services and it could be nice to keep things consistent. Same with the Routing.